data/transactions/logic/eval.go

```diff
@@ -1415,11 +1415,7 @@ func (cx *evalContext) globalFieldToStack(field GlobalField) (sv stackValue, err
 	case LogicSigVersion:
 		sv.Uint = cx.Proto.LogicSigVersion
 	case Round:
-		if (cx.runModeFlags & runModeApplication) == 0 {
```

Comment:
I'd rather fail here with invalid run mode...

Reply:
actually yeah -- I'll use the `not allowed in current mode` error message.

Reply:
what do you think now?